src/utils/documentProducer/variables.ts

```diff
@@ -158,3 +158,25 @@ export const missingRequiredFields = (
     return false;
   });
 };
+
+export const getDependingVariablesStableIdsFromOtherDeliverable = (variablesWithValues: VariableWithValues[]) => {
+  const dependingStableVariablesId = variablesWithValues.reduce((acc: number[], v: VariableWithValues) => {
```

Comment:
This can be a bit simpler if we use Sets

```
const existingStableIds = new Set<string>()
const dependentStableIds = new Set<string>()
variablesWithValues.forEach((variable) => {
  existingStableIds.add(variable.stableId)
  if (variable.dependencyVariableStableId) {
    dependentStableIds.add(variable.dependencyVariableStableId)
  }
});

// cannot use Set.difference here because it is only available is Node 22
existingStableIds.forEach((stableId) => dependentStableIds.remove(stableId));

return Array.from(dependentStableIds) 
```

Reply:
Looks much better! I did a complex thing! Thanks